Sources/ClientRuntime/Telemetry/Metrics/Meter.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// A Meter is the entry point to create the instruments.
+///
+/// The following instrument creation entry points are defined:
+///
+/// - Gauge
+/// - UpDownCounter
+/// - Asynchronous UpDownCounter
+/// - MonotonicCounter
+/// - Asynchronous MonotonicCounter
+/// - Histogram
+public protocol Meter {
+    /// Creates a Gauge, used to measure the current instantaneous value of something.
+    ///
+    /// Examples include: the current memory used by a process
+    ///
+    /// - Parameter name: the instrument name
+    /// - Parameter callback: callback invoked when a Gauge value is read
+    /// - Parameter units: the unit of measure
+    /// - Parameter description: a description of the metric
+    /// - Returns: handle to stop recording Gauge metrics
+    func createGauge(
+        name: String,
+        callback: @escaping (any DoubleAsyncMeasurement) -> Void,
+        units: String?,
+        description: String?
+    ) -> AsyncMeasurementHandle
+    /// Creates an UpDownCounter, used to measure a value that goes up or down.
+    ///
+    /// Examples include: queue length
+    ///
+    /// - Parameter name: the instrument name
+    /// - Parameter units: the unit of measure
+    /// - Parameter description: a description of the metric
+    /// - Returns: an UpDownCounter
+    func createUpDownCounter(
+        name: String,
+        units: String?,
+        description: String?
+    ) -> UpDownCounter
+    /// Creates an asynchronous UpDownCounter, used to measure a value that goes up or down.
+    ///
+    /// Use an asynchronous UpDownCounter instead of a synchronous UpDownCounter when only absolute values are
```

Comment:
Q: What does "absolute values" mean here? If it means mathematical absolute value (i.e., |x|), how does having only positive values affect cost for keeping track of values?

Reply:
Absolute value here does not refer to the mathematical absolute value, but the literal value.

The paradigm is differentiating applying deltas (+/-) and recording literal values.